common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java

```diff
@@ -48,6 +48,13 @@ public long size() {
   }
 
   /**
+   * Reset the size of the memory block.
+   */
```

Comment:
It is dangerous to reset to a invalid size. We should add a check here or put a WARNING in the method comment.

Reply:
Thanks，i will  add a check.